homeassistant/components/climate/tado_v1.py

```diff
@@ -0,0 +1,341 @@
+CONST_OVERLAY_MANUAL = "MANUAL"
+# the temperature will be reset after a timespan
+CONST_OVERLAY_TIMER = "TIMER"
+
+
+# will be used when changing temperature
+CONST_DEFAULT_OPERATION_MODE = CONST_OVERLAY_TADO_MODE
+# will be used when switching to CONST_MODE_OFF
+CONST_DEFAULT_OFF_MODE = CONST_OVERLAY_MANUAL
+
+# DOMAIN = 'tado_v1'
+
+_LOGGER = logging.getLogger(__name__)
+SENSOR_TYPES = ['temperature', 'humidity', 'tado mode', 'power', 'overlay']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the climate platform."""
+    # get the PyTado object from the hub component
+    tado = hass.data['Mytado']
+
+    try:
+        zones = tado.getZones()
+    except RuntimeError:
+        _LOGGER.error("Unable to get zone info from mytado")
+        return False
+
+    tado_data = TadoData(tado)
+
+    climate_devices = []
+    for zone in zones:
+        climate_devices.append(tado_data.create_climate_device(hass,
+                                                               zone['name'],
+                                                               zone['id']))
+
+    if len(climate_devices) > 0:
+        add_devices(climate_devices)
+        tado_data.activate_tracking(hass)
+        return True
+    else:
+        return False
+
+
+class TadoClimate(ClimateDevice):
+    """Representation of a tado climate device."""
+
+    def __init__(self, tado, zone_name, zone_id,
+                 min_temp, max_temp, target_temp, ac_mode,
+                 tolerance=0.3):
+        """Initialization of TadoClimate device."""
+        self._tado = tado
+        self.zone_name = zone_name
+        self.zone_id = zone_id
+
+        self.ac_mode = ac_mode
+
+        self._active = False
+        self._device_is_active = False
+
+        self._cur_temp = None
+        self._cur_humidity = None
+        self._is_away = False
+        self._min_temp = min_temp
+        self._max_temp = max_temp
+        self._target_temp = target_temp
+        self._tolerance = tolerance
+        self._unit = TEMP_CELSIUS
+
+        self._operation_list = [CONST_OVERLAY_MANUAL, CONST_OVERLAY_TIMER,
+                                CONST_OVERLAY_TADO_MODE,
+                                CONST_MODE_SMART_SCHEDULE, CONST_MODE_OFF]
+        self._current_operation = CONST_MODE_SMART_SCHEDULE
+        self._overlay_mode = self._current_operation
+
+    @property
+    def should_poll(self):
+        """
+        No Polling needed for tado climate device.
+
+        because it reuses sensors
+        """
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self.zone_name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def state(self):
+        """
+        Return the current temperature as the state.
+
+        instead of operation_mode.
```

Comment:
But the climate devices return the operation state as the state. That's bad, because (for example) influxdb can't track those items!